File: App/SupabaseConfig.swift
import Foundation

/// Compile-time Supabase configuration. Values committed here are read by
/// `SHJSDSCApp.makeRepository` as the primary source; if either is empty the
/// app falls back to `Bundle.main` (xcconfig path) and then to the offline
/// `DemoRepository`.
///
/// The publishable key is intended for client-side embedding per Supabase
/// docs — it's gated by RLS policies on the server. Never commit a
/// `service_role` key here; that one bypasses RLS.
enum SupabaseConfig {
    static let url: String = "https://khsmwnkitvutcuhypfcz.supabase.co"
    static let anonKey: String = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Imtoc213bmtpdHZ1dGN1aHlwZmN6Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzcyNzUzNDcsImV4cCI6MjA5Mjg1MTM0N30.ZXzw5JmmaGqm8NLurO1pHBBDufmBlYJ8gQnWNVeESCI"
}

File: flutter_app/lib/app/supabase_config.dart
/// Supabase connection config — Flutter counterpart of App/SupabaseConfig.swift.
///
/// The anon (publishable) key is RLS-gated and safe to embed client-side, per
/// Supabase docs — the same key the Swift app commits. Override any of these
/// at build/run time with --dart-define, e.g.
///   flutter run --dart-define=USE_SUPABASE=true
///   flutter run --dart-define=USE_SUPABASE=true \
///     --dart-define=SUPABASE_URL=... --dart-define=SUPABASE_ANON_KEY=...
///
/// [enabled] defaults to false so the offline [DemoRepository] stays the safe
/// default (no network on launch, tests untouched). Flip it on to run the app
/// against the live Postgres backend.
abstract class SupabaseConfig {
  static const bool enabled =
      bool.fromEnvironment('USE_SUPABASE', defaultValue: false);

  static const String url = String.fromEnvironment(
    'SUPABASE_URL',
    defaultValue: 'https://khsmwnkitvutcuhypfcz.supabase.co',
  );

  static const String anonKey = String.fromEnvironment(
    'SUPABASE_ANON_KEY',
    defaultValue:
        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Imtoc213bmtpdHZ1dGN1aHlwZmN6Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzcyNzUzNDcsImV4cCI6MjA5Mjg1MTM0N30.ZXzw5JmmaGqm8NLurO1pHBBDufmBlYJ8gQnWNVeESCI',
  );

  /// True only when the backend is enabled and both values are non-empty.
  static bool get isConfigured =>
      enabled && url.isNotEmpty && anonKey.isNotEmpty;
}
